### PR TITLE
[MAINTENANCE] Remove test_yaml_config and all integration tests that …

### PR DESCRIPTION
**Pull request by tyler-hoffman**
…used it

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: docs/sphinx_api_docs_source/public_api_report.py [hunk 1]
```diff
@@ -109,8 +109,12 @@ def __init__(self, filepath: pathlib.Path, contents: str):
...
     @classmethod
     def create_from_local_file(cls, filepath: pathlib.Path) -> FileContents:
-        with open(filepath) as f:
-            file_contents: str = f.read()
+        try:
+            with open(filepath) as f:
+                file_contents: str = f.read()
+        except Exception as e:
+            breakpoint()
+            print(e)
```
Comment (NathanFarmer) [inline docs/sphinx_api_docs_source/public_api_report.py]: I believe you forgot to remove this

File: tests/datasource/data_connector/test_configured_asset_azure_data_connector.py [hunk 2]
```diff
@@ -33,7 +33,7 @@
 
 @pytest.fixture
 def expected_config_dict():
-    """Used to validate `self_check()` and `test_yaml_config()` outputs."""
```
Comment (NathanFarmer) [inline tests/datasource/data_connector/test_configured_asset_azure_data_connector.py:36]: I'm pretty sure everything related to `self_check()`can also be removed at some point, but that's for another PR. It was all part of this `test_yaml_config()` workflow.

**Review by NathanFarmer — CHANGES_REQUESTED** [review]

File: tests/datasource/data_connector/test_configured_asset_azure_data_connector.py (hunk 2 above)
Comment (tyler-hoffman) [reply tests/datasource/data_connector/test_configured_asset_azure_data_connector.py:36]: Thanks. I had a hunch, but wanted to limit scope!

File: docs/sphinx_api_docs_source/public_api_report.py (hunk 1 above)
Comment (tyler-hoffman) [reply docs/sphinx_api_docs_source/public_api_report.py]: Good catch! Fixed!

**Review by NathanFarmer — APPROVED** [review]

**Pull request merged**